int32 mapgoal
---
int32 res
